service-base/src/test/java/org/eclipse/hono/service/http/TenantTraceSamplingHandlerTest.java

```diff
@@ -0,0 +1,538 @@
+
+        // WHEN handling a request with basic auth for that tenant
+        setupBasicAuthHttpServerRequest(tenantId, authId);
+        tenantTraceSamplingHandler.handle(ctx);
+
+        // THEN the 'sampling.priority' priority was set on the span
+        verify(span).setTag(eq(Tags.SAMPLING_PRIORITY.getKey()), eq(1));
+        verify(ctx).next();
+    }
+
+    /**
+     * Verifies that the handler sets the appropriate span tag for a request with a device auth id for which the trace
+     * sampling mode is set to 'all', the server being configured for single tenant use.
+     */
+    @Test
+    public void testHandleSetsSamplingPriorityForMatchingAuthIdInSingleTenantMode() {
+        // GIVEN a tenant and auth-id for which 'trace-sampling-mode-per-device' is set to 'all'
+        // and a configuration where singleTenant is true.
+        final String tenantId = Constants.DEFAULT_TENANT;
+        final String authId = "testAuthId";
+
+        final TenantObject testTenant = TenantObject.from(tenantId, true);
+        final JsonObject samplingModePerAuthId = new JsonObject();
+        samplingModePerAuthId.put(authId, TraceSamplingMode.ALL.getFieldValue());
+        testTenant.setProperty(TenantConstants.FIELD_TRACE_SAMPLING_MODE_PER_AUTH_ID, samplingModePerAuthId);
+        doAnswer(invocation -> {
+            return Future.succeededFuture(testTenant);
+        }).when(tenantClient).get(eq(tenantId), any(SpanContext.class));
+        config.setSingleTenant(true);
+
+        // WHEN handling a request with basic auth for that tenant
+        setupBasicAuthHttpServerRequest(authId);
+        tenantTraceSamplingHandler.handle(ctx);
+
+        // THEN the 'sampling.priority' priority was set on the span
+        verify(span).setTag(eq(Tags.SAMPLING_PRIORITY.getKey()), eq(1));
+        verify(ctx).next();
+    }
+
+    /**
+     * Verifies that the handler does nothing for a request with a device auth id for which the trace sampling mode is
+     * set to 'default'.
+     */
+    @Test
+    public void testHandleRespectsOverrideForAuthId() {
+        // GIVEN a tenant and auth-id where 'trace-sampling-mode' is set to 'all'
+        // and 'trace-sampling-mode-per-device' is set to 'default'
+        final String tenantId = "testTenant";
+        final String authId = "testAuthId";
+
+        final TenantObject testTenant = TenantObject.from(tenantId, true);
+        testTenant.setProperty(TenantConstants.FIELD_TRACE_SAMPLING_MODE, TraceSamplingMode.ALL.getFieldValue());
+        final JsonObject samplingModePerAuthId = new JsonObject();
+        samplingModePerAuthId.put(authId, TraceSamplingMode.DEFAULT.getFieldValue());
+        testTenant.setProperty(TenantConstants.FIELD_TRACE_SAMPLING_MODE_PER_AUTH_ID, samplingModePerAuthId);
+        doAnswer(invocation -> {
+            return Future.succeededFuture(testTenant);
+        }).when(tenantClient).get(eq(tenantId), any(SpanContext.class));
```

Comment:
just a hint: this can be written like
```
when(tenantClient.get(eq(tenantId), any(SpanContext.class))).thenReturn(Future.succeededFuture(testTenant));
```

which is easier to read (and write) ;-)